### ISSUE TITLE
number of data bearing nodes per cluster

### ISSUE THREAD

**Issue by tajdars** [hazelcast/hazelcast#25668]
In a cluster with mix of data bearing and lite nodes :

I am thinking for redundancy there should be at least 2 data bearing nodes in a cluster ?

If both the data bearing nodes were to go down, partition assignment/reassignment can't occur with lite nodes only ?

Is the partition table replicated on lite nodes ?

If both the data bearing nodes were to go down what happens to the partition table ? Will it be recreated (for new data) when at least one of the data bearing nodes comes back ?

**Comment by vbekiaris** [comment]
> I am thinking for redundancy there should be at least 2 data bearing nodes in a cluster ?

Right, with 2 nodes in the cluster (assuming your data structures have 1 backup as per default), then if one node crashes, backups on the other node will be promoted to primary and partitions will remain available. In general, if your data structures are configured with N backups (and the cluster actually has >= N+1 members), then you can tolerate simultaneous failure of N members (assuming the whole dataset can fit in RAM of remaining members).

> If both the data bearing nodes were to go down, partition assignment/reassignment can't occur with lite nodes only ?

Correct, lite members do not own any partition data, primary or backup.

> Is the partition table replicated on lite nodes ?

Yes, it is known by the lite members of the cluster and they use it to send operations to the appropriate member that owns data for a particular partition.

> If both the data bearing nodes were to go down what happens to the partition table ?

Partition assignments will be empty and all operations targeting partitions (e.g. a `map.get`) will fail since there is no designated partition owner.

> Will it be recreated (for new data) when at least one of the data bearing nodes comes back ?

Exactly.

For these kinds of questions, you can also join [community slack](https://slack.hazelcast.com/), github issues should rather be used to report issues/feature requests in the software.

cheers!
